hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -148,33 +152,72 @@ class RobotHW : public InterfaceManager
     ERROR
   };
 
-  // Return (in realtime) the state of the last doSwitch()
+  /** \brief Return (in realtime) the state of the last doSwitch() */
   virtual SwitchState switchResult() const
   {
     return DONE;
   }
 
-  // Return (in realtime) the state of the last doSwitch() for a given controller
+  /** \brief Return (in realtime) the state of the last doSwitch() for a given controller */
   virtual SwitchState switchResult(const ControllerInfo& /*controller*/) const
   {
     return DONE;
   }
+  /**\}*/
+
+  
+  /** \name Control Loop
+   * 
+   */
+   /**\{*/
 
-  /**
-   * Reads data from the robot HW
+  /** \brief Used to read data from the robot hardware (represented by its interfaces) and populate the raw data.
+   *
+   * The read method is part of the control loop cycle (read, update, write) and used to populate the raw data with 
+   * the state of your robot's hardware resources (joints, sensors, actuators). In your node that handles the control loop,
+   * you should call this method before calling controller_manager::ControllerManager::update and write().
+   * 
+   * The raw data describes regions in memory of your custom robot hardware interface (subclassed from hardware_interface::RobotHW),
+   * which is usually represented by member arrays having names that give semantic meaning (pos, vel, eff) to the registered resources.
+   * The registration of memory regions (assigning pointers) is done via JointStateInterface, for read-only joints,
+   * or JointCommandInterface, for joints that accept commands and provide feedback (read and write), and its subclasses
+   * (eg. VelocityJointInterface) that make use of JointHandle or JointStateHandle through the HardwareResourceManager.
+   * 
+   * \note The name read, of this method, depends on one's directional perspective. 
+   * In the context of the robot hardware_interface::RobotHW it refers to reading state from the robot hardware into the raw data.
+   * This is opposite to the directional perspective where write() is used to fill (write to) 
+   * the raw data with the robot's resource state. Note that this is not the intended approach.
+   * 
```

Comment:
I don't think we need this at all, I think your additions above explain this well enough, but if you think we need the additional clarity let's simplify it to make it even easier to understand:

```suggestion
   * \note The name 'read()' refers to reading state from the hardware.
   * This complements 'write()', which refers to writing commands to the hardware.
   * 
```

Reply:
I'd prefer to keep it the way you suggested and don't remove it completely. Just for extra clarity.